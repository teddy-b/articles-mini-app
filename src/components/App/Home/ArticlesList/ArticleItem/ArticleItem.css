.article-item {
  padding: 1em;
  display: flex;
}

.article-item .article-item-img {
  margin-right: 1em;
  width: 40%;
}

.article-item .article-item-img img {
  width: 100%;
}

.article-item .article-item-content {
  width: 60%;
  max-height: 30vh;
}

.article-item .article-item-content .article-item-title {
  font-weight: bold;
}

.article-item .article-item-text {
  overflow: hidden;
  position: relative; 
  line-height: 1.2em;
  max-height: 3.6em; 
  text-align: justify;
  margin-right: -1em;
  padding-right: 1em;
}
.article-item .article-item-text:before {
  content: '...';
  position: absolute;
  right: 0;
  bottom: 0;
}
.article-item .article-item-text:after {
  content: '';
  position: absolute;
  right: 0;
  width: 1em;
  height: 1em;
  margin-top: 0.2em;
  background: white;
}

.article-item .article-item-comments {
  color: blue;
  text-decoration: underline;
}

.article-item:hover {
  cursor: pointer;
  background-color: gray;
}

.article-item.selected,
.article-item:hover .article-item-text:after,
.article-item.selected .article-item-text:after {
  background-color: gray;
}
