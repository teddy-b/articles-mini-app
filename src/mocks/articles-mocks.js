export const articlesMock = [
  {
    title: 'CERN experiment discovers five new particles',
    text: 'A new group of particles which have long been hiding in plain sight have finally been discovered thanks to the incredibly sensitive LHCb experiment at CERN. LHCb, otherwise known as the Large Hadron Collider beauty experiment, is one of seven experiments collecting data at the Large Hadron Collider at CERN. UK participation in the experiment is funded by STFC.',
    imageUrl: 'http://www.stfc.ac.uk/stfc/cache/file/4821074D-0D31-4964-9318CA0AE81E0639.jpg',
    id: 0,
    createdAt: 1530698004815,
    commentsCount: 8
  },
  {
    title: 'Show HN: Marta – A fast and minimalistic file manager for macOS',
    text: 'File Manager for macOS. Native. Minimalistic. Fast. ',
    imageUrl: 'https://marta.yanex.org/data/images/dark.png',
    id: 1,
    createdAt: 1530698004815,
    commentsCount: 7
  },
  {
    title: 'Tears',
    text: 'So humans are a peculiar species. You probably kne…specific traits and behaviors are uniquely human.',
    imageUrl: 'http://www.meltingasphalt.com/wp-content/uploads/2014/12/weeping_equation.png',
    id: 2,
    createdAt: 1530698004815,
    commentsCount: 0
  }
];

export default articlesMock;
