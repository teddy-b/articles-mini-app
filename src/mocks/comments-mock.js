export const commentsMock = [
  {
    articleId: 0,
    author: 'grabcocque',
    createdAt: 1530698004815,
    id: 0,
    parentCommentId: null,
    repliesCount: 0,
    text: 'I have to admit, I rather admire their putting a brave face on things, with their mildly-too-insistent claim that the LHC is vital for finding new physics despite the increasing likelihood it\'s not going to find any. In that sense the LHC is a political failure, because it has failed at its primary political job, which is to make the argument for an even bigger collider.'
  },
  {
    articleId: 0,
    author: 'teddythetwig',
    createdAt: 1530698004815,
    id: 1,
    parentCommentId: null,
    repliesCount: 2,
    text: 'I assume these particles were already predicted as part of the Standard Model?'
  },
  {
    articleId: 0,
    author: 'guitarbill',
    createdAt: 1530698004815,
    id: 6,
    parentCommentId: null,
    repliesCount: 0,
    text: 'AFAIK, the experimentally determined rest masses of these excited states/particles (same thing, different was of looking at it) agree with the calculated ones well. So yes, they were predicted. No surprises sadly; of course it\'s still a huge achievement!'
  }
];

export default commentsMock;
