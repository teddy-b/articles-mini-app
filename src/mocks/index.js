export * from './articles-mocks';
export * from './comments-mock';
export * from './replies-mock';
export * from './replies-replies-mock';
