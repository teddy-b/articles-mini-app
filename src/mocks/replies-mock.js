export const repliesMock = [
  {
    articleId: 0,
    author: 'brutuscat',
    createdAt: 1530698004815,
    id: 2,
    parentCommentId: 1,
    repliesCount: 0,
    text: 'Professor Tim Gershon, Professor of Physics at University of Warwick and UK spokesperson for the LHCb experiment: '
  },
  {
    articleId: 0,
    author: 'gpderetta',
    createdAt: 1530698004815,
    id: 3,
    parentCommentId: 1,
    repliesCount: 2,
    text: 'yes, and note that these are not fundamental particles (like the Higgs was for example), but composite particles of yet another combination of the fundamental quarks. The SM predicts the existence of hundreds (thousands?) of these.'
  }
];

export default repliesMock;
