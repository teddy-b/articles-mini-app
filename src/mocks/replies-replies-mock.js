export const repliesRepliesMock = [
  {
    articleId: 0,
    author: 'inlineint',
    createdAt: 1530698004815,
    id: 4,
    parentCommentId: 3,
    repliesCount: 0,
    text: 'Actually it is even a one new particle (Omega_c baryon), but they observed five different excited energy states of it (like observing different excited states of a Hydrogen atom), so called resonances [1]. But the discovery is still exciting because it should have been really hard to find something that we don\'t really know how looks like in such large amount of noise.'
  },
  {
    articleId: 0,
    author: 'binarymax',
    createdAt: 1530698004815,
    id: 5,
    parentCommentId: 3,
    repliesCount: 0,
    text: 'Because a prediction is just an assumption (theory), and it can become a house of cards when basing future science on that assumption. Observation is proof, so future science can use that proof without worry.'
  }
];

export default repliesRepliesMock;
