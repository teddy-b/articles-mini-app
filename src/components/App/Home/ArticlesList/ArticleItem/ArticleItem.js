import React from 'react';
import { Link } from 'react-router-dom';

import './ArticleItem.css';

const ArticleItem = ({
  commentsCount,
  id,
  imageUrl,
  text,
  title,
  onLoadComments,
  onSelectArticle,
  selected
}) => (
  <Link
    className="styled-link"
    to={`/article/${id}`}
    onClick={() => {
      onSelectArticle(id);
      onLoadComments(id);
    }}
  >
    <div className={`article-item ${selected ? 'selected' : ''}`}>
      <div className="article-item-img">
        <img src={imageUrl} alt={title} />
      </div>
      <div className="article-item-content">
        <div className="article-item-title">{title}</div>
        <div className="article-item-text">{text}</div>
        <div className="article-item-comments">{commentsCount} comments</div>
      </div>
    </div>
  </Link>
);

export default ArticleItem;
