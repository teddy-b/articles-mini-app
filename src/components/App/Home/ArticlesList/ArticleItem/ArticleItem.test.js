import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import ArticleItem from './ArticleItem';
import { articlesMock } from '../../../../../mocks';

describe('ArticleItem', () => {
  const onLoadCommentsMock = jest.fn();
  const onSelectArticleMock = jest.fn();
  let selectedMock = false;

  const props = {
    ...articlesMock[0],
    onLoadComments: onLoadCommentsMock,
    onSelectArticle: onSelectArticleMock,
    selected: selectedMock
  };

  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <ArticleItem
          {...props}
        />
      </MemoryRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly while selected', () => {
    selectedMock = true;
    const component = renderer.create(
      <MemoryRouter>
        <ArticleItem
          {...props}
          selected={selectedMock}
        />
      </MemoryRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('select article and load comments on click', () => {
    const component = mount(
      <MemoryRouter>
        <ArticleItem
          {...props}
        />
      </MemoryRouter>
    );

    component.find('Link').simulate('click');

    expect(onLoadCommentsMock).toHaveBeenCalled();
    expect(onSelectArticleMock).toHaveBeenCalled();
  });
});
