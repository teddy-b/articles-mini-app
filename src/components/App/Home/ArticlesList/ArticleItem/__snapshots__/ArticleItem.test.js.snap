// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem renders correctly 1`] = `
<a
  className="styled-link"
  href="/article/0"
  onClick={[Function]}
>
  <div
    className="article-item "
  >
    <div
      className="article-item-img"
    >
      <img
        alt="CERN experiment discovers five new particles"
        src="http://www.stfc.ac.uk/stfc/cache/file/4821074D-0D31-4964-9318CA0AE81E0639.jpg"
      />
    </div>
    <div
      className="article-item-content"
    >
      <div
        className="article-item-title"
      >
        CERN experiment discovers five new particles
      </div>
      <div
        className="article-item-text"
      >
        A new group of particles which have long been hiding in plain sight have finally been discovered thanks to the incredibly sensitive LHCb experiment at CERN. LHCb, otherwise known as the Large Hadron Collider beauty experiment, is one of seven experiments collecting data at the Large Hadron Collider at CERN. UK participation in the experiment is funded by STFC.
      </div>
      <div
        className="article-item-comments"
      >
        8
         comments
      </div>
    </div>
  </div>
</a>
`;

exports[`ArticleItem renders correctly while selected 1`] = `
<a
  className="styled-link"
  href="/article/0"
  onClick={[Function]}
>
  <div
    className="article-item selected"
  >
    <div
      className="article-item-img"
    >
      <img
        alt="CERN experiment discovers five new particles"
        src="http://www.stfc.ac.uk/stfc/cache/file/4821074D-0D31-4964-9318CA0AE81E0639.jpg"
      />
    </div>
    <div
      className="article-item-content"
    >
      <div
        className="article-item-title"
      >
        CERN experiment discovers five new particles
      </div>
      <div
        className="article-item-text"
      >
        A new group of particles which have long been hiding in plain sight have finally been discovered thanks to the incredibly sensitive LHCb experiment at CERN. LHCb, otherwise known as the Large Hadron Collider beauty experiment, is one of seven experiments collecting data at the Large Hadron Collider at CERN. UK participation in the experiment is funded by STFC.
      </div>
      <div
        className="article-item-comments"
      >
        8
         comments
      </div>
    </div>
  </div>
</a>
`;
